main.go

```diff
@@ -43,9 +51,58 @@ func main() {
 
 	err = tf5server.Serve(tfeProviderName, func() tfprotov5.ProviderServer {
 		return mux.Server()
-	})
+	}, withServeOptions(ctx, debugFlag)...)
 	if err != nil {
 		log.Printf("[ERROR] Could not start serving the ProviderServer: %v", err)
 		os.Exit(1)
 	}
 }
+
+func withServeOptions(ctx context.Context, debugFlag *bool) []tf5server.ServeOpt {
+	serveOpts := []tf5server.ServeOpt{}
+	if *debugFlag {
+		reattachConfigCh := make(chan *plugin.ReattachConfig)
+		go func() {
+			reattachConfig, err := waitForReattachConfig(reattachConfigCh)
+			if err != nil {
+				fmt.Printf("Error getting reattach config: %s\n", err)
+				return
+			}
+			printReattachConfig(reattachConfig)
+		}()
+		serveOpts = append(serveOpts, tf5server.WithDebug(ctx, reattachConfigCh, nil))
```

Comment:
Can we add a TODO comment to update this to `tf5server.WithManagedDebug()` when we upgrade `terraform-plugin-go` to `v0.6.0`? Also mention that we can remove the custom reattach config lifecycle.